### ISSUE TITLE
AI rename

### ISSUE THREAD

**Issue by DerKorb** [getcursor/cursor#897]
In vscode you can rename by pressing F2. 
It will aplly your rename throughout the codebase. 

What I would like to see is an AI rename the same way.
Often times when coding you need to come up with good names for variables. Sometimes this can make you stuck for a bit, while figuring out what the variable will do while coming up with algorithms. 
What I often do in these situations is just calling it something random, writing the algorithm and then naming it when I have a better picture of how the algorithm works. For this usecase it would be awesome to just press the <AI rename> button on the variable similiar how F2 does, but it would prompt gpt with "What is the best variable name for X in this algorithm".

Awesome IDE btw, this does so much I am trying to build my self for quite some while now. It would be super awesome if you could run multiple AI edits in paralell so you could jump between places while waiting for gpt to finish the edit.  Could even show a similiar blip like for unsaved files when the edit is ready.

**Comment by sublimator** [comment]
> It would be super awesome if you could run multiple AI edits in paralell so you could jump between places while waiting for gpt to finish the edit. Could even show a similiar blip like for unsaved files when the edit is ready.

Yep! Until the point where humans are completely obviated, manual context editing (at the least a raw "view" for debugging ) and overseer of multiple agents seems the "future" (however long that will last ... )

**Comment by jakobFNF** [comment]
I like this variable name suggestion idea.

**Comment by YoshiYoshi360** [comment]
This would save a lot of tokens

**Comment by pahen** [comment]
Any news on this? The only thing i miss from VSCode.

**Comment by leandroribeir0** [comment]
I miss that feature from VSCode too.

**Comment by Yarden92** [comment]
almost 3 years now, and we are still waiting for this existing feature..
https://devblogs.microsoft.com/visualstudio/ai-powered-rename-suggestions/?utm_source=chatgpt.com

**Comment by leandroribeir0** [comment]
Guys, pleaseeee...
